Lib/idlelib/calltips.py

```diff
@@ -145,10 +148,16 @@ def get_argspec(ob):
     else:
         fob = ob
     if isinstance(fob, (types.FunctionType, types.MethodType)):
-        argspec = inspect.formatargspec(*inspect.getfullargspec(fob))
+        try:
+            argspec = str(inspect.signature(fob))
+        except ValueError:
+            argspec = "This function has an invalid method signature"
+            return argspec
+
         if (isinstance(ob, (type, types.MethodType)) or
                 isinstance(ob_call, types.MethodType)):
-            argspec = _first_param.sub("", argspec)
+            if argspec.startswith("(self,"):
```

Comment:
Yes, you are right. I'll still need to figure out how it does different from `getfullargspec`